source/MaterialXCore/Interface.h

```diff
@@ -247,6 +247,12 @@ class MX_CORE_API Input : public PortElement
     /// Return the node, if any, to which this input is connected.
     NodePtr getConnectedNode() const override;
 
+    /// Set the input interface. If the interface name specified is not empty,
+    /// the the interface is set if a corresponding input exists in the graph or document
+    /// containing the input's node. If the interface name is empty then the 
+    /// interface reference removed.
+    void setInterfaceInput(const string& interfaceName);
```

Comment:
The proposed name and documentation for this method seem very unclear to me, and it's my sense that it would confuse a new developer.

Maybe what you really want is to refactor the existing `setInterfaceName` method, allowing it to both set the interface name string and clear all previous bindings?  I see that this method is currently on `ValueElement`, so that it's available for `Token` elements as well, but this current structure doesn't need to remain as it stands, and you could potentially move it to any appropriate level of the class hierarchy to achieve the goal of an improved `setInterfaceName` method that works as expected for all classes.